### PR TITLE
bug fix for bonding.py

### PR DESCRIPTION
**Pull request by vaishnavibhat**
- using bringup() from interface utility
- deleting ifcfg-bond file post test created during setup

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vaishnavibhat** [comment]
avocado run bonding.py -m bonding.py.data/bonding_single.yaml 
JOB ID     : 96f10751e8fa4f2b9434e4afc385ed382ad1bdca
JOB LOG    : /root/bond_scen/upstream/tests/results/job-2020-11-11T22.36-96f1075/job.log
 (1/3) bonding.py:Bonding.test_setup;run-385e: PASS (20.94 s)
 (2/3) bonding.py:Bonding.test_run;run-385e: PASS (238.16 s)
 (3/3) bonding.py:Bonding.test_cleanup;run-385e: PASS (9.02 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/bond_scen/upstream/tests/results/job-2020-11-11T22.36-96f1075/results.html
JOB TIME   : 268.94 s

[bonding.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5525645/bonding.log)

File: io/net/bonding.py [hunk 1]
```diff
@@ -470,8 +459,17 @@ def test_cleanup(self):
                     self.log.warn("unable to bring to original state in peer")
                 time.sleep(self.sleep_time)
         self.error_check()
+
+        try:
+            NetworkInterface(self.bond_name, self.localhost).restore_from_backup()
```
Comment (abdhaleegit) [inline io/net/bonding.py:464]: .restore_from_backup() should be called only when we have .saved() and .bring_up() has been called.. @smruti77 Please clarify here.

I mean if there is no sysfs bond file saved, than  why restore backup.

File: io/net/bonding.py [hunk 2]
```diff
@@ -470,8 +459,17 @@ def test_cleanup(self):
...
+            self.log.info("backup file not availbale, could not restore file.")
         for host_interface in self.host_interfaces:
             networkinterface = NetworkInterface(host_interface, self.localhost)
+            try:
+                networkinterface.bring_up()
+            except Exception:
+                self.fail("Interface is taking long time to link up")
             if networkinterface.set_mtu("1500") is not None:
                 self.cancel("Failed to set mtu in host")
             try:
```
Comment (abdhaleegit) [inline io/net/bonding.py:475]: Can we run the above changes, with and with out a sysfs bonding configuration file..   ?

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
@vaishnavibhat I have few clarifications requested, please have a look .

File: io/net/bonding.py (hunk 1 above)
Comment (abdhaleegit) [reply io/net/bonding.py:464]: also we are not creating sysfs bond configuration file in the bond_setup test ...

Comment (vaishnavibhat) [reply io/net/bonding.py:464]: We have a ifcfg-bond0 file created for all our bonding tests. It is defined in 
 def setup_ip(self)

We are creating bond master and slave using sysfs but the test flow ends up having a ifcfg-bond0 file created everytime. Also the format of this ifcfg-bond0 is like any other ifcfg-eth* file.

File: io/net/bonding.py (hunk 2 above)
Comment (vaishnavibhat) [reply io/net/bonding.py:475]: yes, it works for with and without a sysfs bonding file.

**Comment by vaishnavibhat** [comment]
avocado run bonding.py -m bonding_mode1.yaml 

JOB ID     : 22e0ff6f9f94445d8f99ca0a136164521c2dc225
JOB LOG    : /root/lates/avocado-fvt-wrapper/results/job-2020-11-24T23.05-22e0ff6/job.log
 (1/3) bonding.py:Bonding.test_setup;run-b2da: PASS (20.84 s)
 (2/3) bonding.py:Bonding.test_run;run-b2da: PASS (134.29 s)
 (3/3) bonding.py:Bonding.test_cleanup;run-b2da: PASS (7.49 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/lates/avocado-fvt-wrapper/results/job-2020-11-24T23.05-22e0ff6/results.html
JOB TIME   : 163.45 s

[Uploading bond_fix.log…]()

**Comment by vaishnavibhat** [comment]
I am doing some changes to the patch sent previously. I am reverting the changes .restore_from_backup() used on bond interface since sysfs bond creation is not creating ifcfg-bond0 file. 
This issue was particular to the bonding scenario that we have. After running the scenario we have ifcfg-bond0 created but not removed. This ifcfg-bond0 file is created by the htx code and will do the changes to remove the ifcfg  file in the respective script.

**Comment by abdhaleegit** [comment]
@vaishnavibhat as you have reverted the old changes, could you also edit the commit message ? if there is nothing else here this LGTM

**Review by abdhaleegit — APPROVED** [review]

**Review by narasimhan-v — APPROVED** [review]
@abdhaleegit Thanks for the review

**Pull request merged**